src/Swashbuckle.AspNetCore.SwaggerGen/XmlComments/XmlCommentsSchemaFilter.cs

```diff
@@ -61,7 +62,53 @@ private void ApplyPropertyComments(OpenApiSchema propertySchema, MemberInfo memb
 
             var exampleNode = memberNode.SelectSingleNode(ExampleXPath);
             if (exampleNode != null)
-                propertySchema.Example = new OpenApiString(XmlCommentsTextHelper.Humanize(exampleNode.InnerXml));
+            {
+                var exampleString = XmlCommentsTextHelper.Humanize(exampleNode.InnerXml);
+                var memberType = (memberInfo.MemberType & MemberTypes.Field) != 0 ? ((FieldInfo) memberInfo).FieldType : ((PropertyInfo) memberInfo).PropertyType;
+                propertySchema.Example = ConvertToOpenApiType(exampleString, memberType);
+            }
+        }
```

Comment:
There's already a class called `OpenApiPrimitiveFactory` that you should be able to re-use in place of the `ConvertToOpenApiType` method. I _think_ something along the following lines will work:

```csharp
var typedExample = TypeDescriptor.GetConverter(memberType).ConvertFrom(exampleString);
propertySchema.Example = OpenApiPrimitiveFactory.CreateFrom(typedExample)
```

Reply:
@domaindrivendev Great catch! Fixed in 7a5ebf7.